src/QMCHamiltonians/NonLocalECPotential.deriv.cpp

```diff
@@ -137,26 +137,39 @@ NonLocalECPComponent::RealType NonLocalECPComponent::evaluateValueAndDerivatives
     for (int j = 0; j < nknot; j++)
     {
       PosType pos_now = W.R[iel];
+      //W.makeMoveAndCheck(iel, deltarV[j]);
+
+//#if defined(QMC_COMPLEX)
+//      psiratio[j] = psi.ratio(W, iel) * ValueType( std::cos(psi.getPhaseDiff()), std::sin(psi.getPhaseDiff()) );
+//#else
+//      psiratio[j] = psi.ratio(W, iel);
+//#endif
+//      psiratio[j] = psi.full_ratio(W, iel);
       W.makeMove(iel, deltarV[j]);
-#if defined(QMC_COMPLEX)
-      psiratio[j] = psi.ratio(W, iel) * std::cos(psi.getPhaseDiff());
-#else
-      psiratio[j] = psi.ratio(W, iel);
-#endif
+//#if defined(QMC_COMPLEX)
+//      psiratio[j] = psi.ratio(W, iel) * std::cos(psi.getPhaseDiff());
+//#else
+//      psiratio[j] = psi.ratio(W, iel);
+//#endif
+      psiratio[j] = psi.full_ratio(W, iel);
       psi.resetPhaseDiff();
 
       //use existing methods
       W.acceptMove(iel);
+      psi.acceptMove(W, iel);
 
       std::fill(dlogpsi_t.begin(), dlogpsi_t.end(), 0.0);
-      std::fill(dlogpsi_ct.begin(), dlogpsi_ct.end(), 0.0);
-      psi.evaluateDerivatives(W, optvars, dlogpsi_ct, dhlogpsi_ct);
+      //std::fill(dlogpsi_ct.begin(), dlogpsi_ct.end(), 0.0);
+      //psi.evaluateDerivatives(W, optvars, dlogpsi_t, dhlogpsi_t);
+      psi.evaluateDerivativesWF(W, optvars, dlogpsi_t);
       for (int v = 0; v < dlogpsi_t.size(); ++v)
-        dratio(v, j) = std::real(dlogpsi_t[v]);
+        dratio(v, j) = dlogpsi_t[v];
 
       PosType md = -1.0 * deltarV[j];
       W.makeMove(iel, md);
       W.acceptMove(iel);
+      ValueType tmp_ratio = psi.full_ratio(W, iel);
+      psi.acceptMove(W, iel);
```

Comment:
Following the above call sequence? Do we have sufficient test? A test with Jastrow should catch this.

Reply:
I think both way works, and I have changed it to follow the above call sequence.

Reply:
The exact call sequence is required. It works for you probably because your wavefunction is simple.

Reply:
I have changed it and made sure to use the exact same call sequence.